src/main/resources/schema/elytron-client-1_1.xsd

```diff
@@ -204,43 +527,152 @@
+            </xsd:documentation>
+        </xsd:annotation>
         <!-- Protection parameter -->
         <xsd:choice minOccurs="0" maxOccurs="1">
             <xsd:element name="key-store-credential" type="key-store-ref-type" minOccurs="1" maxOccurs="1"/>
             <xsd:element name="key-store-clear-password" type="clear-password-type" minOccurs="1" maxOccurs="1"/>
             <xsd:element name="credential-store-reference" type="credential-store-reference-type" minOccurs="1" maxOccurs="1"/>
         </xsd:choice>
-        <xsd:attribute name="key-store-name" type="xsd:string" use="required"/>
-        <xsd:attribute name="alias" type="xsd:string" use="optional"/>
+        <xsd:attribute name="key-store-name" type="xsd:string" use="required">
+            <xsd:annotation>
+                <xsd:documentation>
+                    Name of the KeyStore being referenced to load the entry from.
+                </xsd:documentation>
+            </xsd:annotation>
+        </xsd:attribute>
+        <xsd:attribute name="alias" type="xsd:string" use="optional">
+            <xsd:annotation>
+                <xsd:documentation>
+                    The alias of the entry to load from the referenced KeyStore, 
+                    this can only be omitted for KeyStores that contain only a single entry.
+                </xsd:documentation>
+            </xsd:annotation>
+        </xsd:attribute>
     </xsd:complexType>
 
     <xsd:complexType name="credential-store-reference-type">
-        <xsd:attribute name="store" type="xsd:string"/>
-        <xsd:attribute name="alias" type="xsd:string"/>
-        <xsd:attribute name="clear-text" type="xsd:string"/>
+        <xsd:annotation>
+            <xsd:documentation>
+                Reference to a credential stored in a credential store.
+            </xsd:documentation>
+        </xsd:annotation>
+        <xsd:attribute name="store" type="xsd:string">
+            <xsd:annotation>
+                <xsd:documentation>
+                    Credential store name.
+                </xsd:documentation>
+            </xsd:annotation>
+        </xsd:attribute>
+        <xsd:attribute name="alias" type="xsd:string">
+            <xsd:annotation>
+                <xsd:documentation>
+                    Alias in the credential store.
+                </xsd:documentation>
+            </xsd:annotation>
+        </xsd:attribute>
+        <xsd:attribute name="clear-text" type="xsd:string">
+            <xsd:annotation>
+                <xsd:documentation>
+                    Credential store password in clear text.
+                </xsd:documentation>
+            </xsd:annotation>
+        </xsd:attribute>
     </xsd:complexType>
 
     <xsd:complexType name="trust-manager-type">
-        <xsd:attribute name="provider-name" type="xsd:string"/>
-        <xsd:attribute name="algorithm" type="xsd:string"/>
+        <xsd:annotation>
+            <xsd:documentation>
+                Trust manager definition.
```

Comment:
Trust manager definition. Decides whether credentials presented by a peer should be accepted.